src/yaml/yaml.cr

```diff
@@ -14,7 +15,34 @@ module YAML
   alias Type = String | Hash(Type, Type) | Array(Type) | Nil
   alias EventKind = LibYAML::EventType
 
-  def self.load(data)
+# Serializes a document in YAML format.
+#
+# ```yaml
+# # ./foo.yml
+# foo:
+#   bar: "foobar"
+# ```
+#
+# Returns `Hash` if *data* is in YAML format.
+# ```crystal 
+# require "yaml"
+#
+# YAML.load(File.read("./foo.yml"))
+# #=> {"foo" => {"bar" => "foobar"}}
+# ```
+#
+# Returns the same value if *data* is not in YAML format.
+# ```crystal 
+# YAML.load("foo")
+# #=> "foo"
+# ```
+#
+# Returns `Nil` if *data* is a blank string.
+# ```crystal 
+# YAML.load("")
+# #=> nil
```

Comment:
Again the absence of any value is a valid YAML document and just is represented with nil, compare

```
foo:
  bar:
  baz:
    -
    - quux
```